src/paths/random_walk.c

```diff
@@ -301,3 +240,152 @@ igraph_error_t igraph_random_edge_walk(const igraph_t *graph,
+                      "number of vertices in the graph (%" IGRAPH_PRId
+                      "), got %" IGRAPH_PRId ".", IGRAPH_EINVAL,
+                      vc, start);
+    }
+    if (steps < 0) {
+        IGRAPH_ERRORF("Number of steps should be non-negative, got %"
+                      IGRAPH_PRId ".", IGRAPH_EINVAL, steps);
+    }
+
+    if (weights) {
+        if (igraph_vector_size(weights) != ec) {
+            IGRAPH_ERROR("Invalid weight vector length.", IGRAPH_EINVAL);
+        }
+        if (ec > 0) {
+            igraph_real_t min = igraph_vector_min(weights);
+            if (min < 0) {
+                IGRAPH_ERROR("Weights must be non-negative.", IGRAPH_EINVAL);
+            }
+            else if (igraph_is_nan(min)) {
+                IGRAPH_ERROR("Weights must not contain NaN values.", IGRAPH_EINVAL);
+            }
+        }
+    }
+
+    if (edges || weights) {
+        return igraph_i_random_walk_inclist(graph, weights, vertices, edges,
+                                            start, mode, steps, stuck);
+    }
+    else {
+        return igraph_i_random_walk_adjlist(graph, vertices,
+                                            start, mode, steps, stuck);
+    }
+}
+
+
+/**
+ * \function igraph_random_edge_walk
+ * \brief Perform a random walk on a graph and return the traversed edges
+ *
+ * Performs a random walk with a given length on a graph, from the given
+ * start vertex. Edge directions are (potentially) considered, depending on
+ * the \p mode argument.
+ *
+ * \param graph The input graph, it can be directed or undirected.
+ *   Multiple edges are respected, so are loop edges.
+ * \param weights A vector of non-negative edge weights. It is assumed
+ *   that at least one strictly positive weight is found among the
+ *   outgoing edges of each vertex. Additionally, no edge weight may
+ *   be NaN. If either case does not hold, an error is returned. If it
+ *   is a NULL pointer, all edges are considered to have equal weight.
+ * \param edgewalk An initialized vector; the indices of traversed
+ *   edges are stored here. It will be resized as needed.
+ * \param start The start vertex for the walk.
+ * \param steps The number of steps to take. If the random walk gets
+ *   stuck, then the \p stuck argument specifies what happens.
+ * \param mode How to walk along the edges in directed graphs.
+ *   \c IGRAPH_OUT means following edge directions, \c IGRAPH_IN means
+ *   going opposite the edge directions, \c IGRAPH_ALL means ignoring
+ *   edge directions. This argument is ignored for undirected graphs.
+ * \param stuck What to do if the random walk gets stuck.
+ *   \c IGRAPH_RANDOM_WALK_STUCK_RETURN means that the function returns
+ *   with a shorter walk; \c IGRAPH_RANDOM_WALK_STUCK_ERROR means
+ *   that an error is reported. In both cases, \p edgewalk is truncated
+ *   to contain the actual interrupted walk.
+ *
+ * \return Error code.
+ *
+ * \deprecated-by igraph_strvector_set_len 0.10.0
```

Comment:
```suggestion
 * \deprecated-by igraph_random_walk 0.10.0
```

Reply:
Oh, Sorry for this. I was referring to some other PRs to see how the functions are deprecated and forgot to change this. I've corrected it.

Reply:
No problem, that happens...